src/language-markdown/printer-markdown.js

```diff
@@ -357,12 +357,27 @@ function genericPrint(path, options, print) {
                 align(
                   " ".repeat(options.tabWidth),
                   printChildren(path, options, print, {
-                    processor: (childPath, index) =>
-                      index === 0
-                        ? group(
-                            concat([softline, softline, childPath.call(print)])
-                          )
-                        : childPath.call(print)
+                    processor: (childPath, index) => {
+                      const node = path.getValue();
+                      const isMultilineFootnote =
+                        node.children &&
+                        node.children[0].children &&
+                        node.children[0].children.some(
+                          child => child.value === "\n"
+                        );
+
+                      return index === 0
+                        ? isMultilineFootnote
+                          ? group(concat([softline, childPath.call(print)]))
+                          : group(
+                              concat([
+                                softline,
+                                softline,
```

Comment:
Why do we need two softlines here?

Reply:
I don't know why but I think the same reason that softline is inserted between headers such like `#`  🤔

Reply:
I realized that it caused the issue, so I remove it.
Thanks for pointed out!